src/playwright/merge-coverage-reports.ts

```diff
@@ -0,0 +1,174 @@
+import { readFileSync } from 'node:fs';
+import path from 'node:path';
+import istanbulLibCoverage, {
+  type CoverageMapData,
+} from 'istanbul-lib-coverage';
+import { createContext, type ReportNode } from 'istanbul-lib-report';
+import { globby } from 'globby';
+import { create, type LinkMapper, type JsonOptions } from 'istanbul-reports';
+import type { FullConfig } from '@playwright/test';
+import playwrightConfiguration from './playwright.config.js';
+import { type CoverageReporterOptions } from './coverage-reporter.js';
+
+// 1. Gather the information we need from Playwright's configuration.
```

Comment:
Numbering only goes so far. Open to suggestions for making this file easier to read.